Source/WebCore/page/make-user-agent-string-quirks.py

```diff
@@ -0,0 +1,178 @@
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+
+user_agents = {
+    'macos': {
+        'chrome': "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"
+    }
+}
+
+
+def lowercase_first_letter(string):
+    return string[0].lower() + string[1:] if len(string) and string[1].islower() else string
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate UserAgentOverride.h')
+    parser.add_argument('--input')
+    args = parser.parse_args()
+
+    with open(args.input, 'r', encoding='utf-8') as input_file:
+        overrides = json.load(input_file)
+
+    with open('UserAgentStringOverrides.h', 'w') as output_file:
+        def write(text):
+            output_file.write(text)
+
+        def writeln(text):
+            write(text + '\n')
+
+        writeln('''/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/RobinHoodHashMap.h>
+
+namespace WebCore {
+
+class UserAgentStringOverrides {
+    WTF_MAKE_NONCOPYABLE(UserAgentStringOverrides); WTF_MAKE_FAST_ALLOCATED;
+public:
+    UserAgentStringOverrides();
+    std::optional<String> getUserAgentStringOverrideForDomain(StringView domain) const;
+
+private:
+    enum class UserAgentString : uint8_t {''')
+        for platform in sorted(user_agents.keys()):
+            for user_agent in user_agents[platform]:
+                entry = platform.upper() + "_" + user_agent.upper() + ","
+            writeln("        " + entry)
```

Comment:
This makes me a bit nervous. It should probably be a function. And then we could create a unit test to make double sure, only the right format is generated.